apis/validator/block.v3.yaml

```diff
@@ -76,14 +76,14 @@ get:
                 example: "12345"
               data:
                 oneOf:
-                 - $ref: '../../beacon-node-oapi.yaml#/components/schemas/BeaconBlock'
+                 - $ref: "../../beacon-node-oapi.yaml#/components/schemas/BeaconBlock"
                  - $ref: "../../beacon-node-oapi.yaml#/components/schemas/Altair.BeaconBlock"
                  - $ref: "../../beacon-node-oapi.yaml#/components/schemas/Bellatrix.BeaconBlock"
                  - $ref: "../../beacon-node-oapi.yaml#/components/schemas/Bellatrix.BlindedBeaconBlock"
                  - $ref: "../../beacon-node-oapi.yaml#/components/schemas/Capella.BeaconBlock"
                  - $ref: "../../beacon-node-oapi.yaml#/components/schemas/Capella.BlindedBeaconBlock"
                  - $ref: "../../beacon-node-oapi.yaml#/components/schemas/Deneb.BlockContents"
```

Comment:
remove this blockContents too?

Reply:
This is where it's a little confusing and asymmetric. `BlockContents` is still used in the unblinded flow, because in order to keep it stateless we need to pass blobs along with the block back and forth. So this endpoint can return `Deneb.BlockContents` or `Deneb.BlindedBeaconBlock`

Reply:
i would prefer a separate call by validator to pull blobs if it plans to reroute the publish since the data transfer is non trivial and will also resolve code spaghetti in the api, to address latency, validator call concurrently pull blobs while its signing the block

Reply:
That would not meet the requirement for statelessness, as it would be possible for the beacon node to exit between returning the data in this call and a separate request being made for the blobs.

Reply:
>  i would prefer a separate call by validator to pull blobs if it plans to reroute the publish since the data transfer is non trivial and will also resolve code spaghetti in the api, to address latency, validator call concurrently pull blobs while its signing the block

I'm not sure this can be done without reintroducing the state requirement. Potentially we could set up an avenue for BNs to query ELs to check their local mempool for blobs matching a commitment, but I don't think this is really a great solution either. It might be better to mitigate the expense of transferring blob data by preferring SSZ transfer or preferring the blinded flow even when not using an external builder (this does put you at risk of missing a block proposal if you are using fallback between BNs however).

Reply:
beacon nodes are stateful, i am not sure why we are imposing the requirement of statefulness

Reply:
I think it's been made a requirement to support both specifically in block production to support a greater variety of setups and make sure the proposal flow is generally more reliable

Reply:
if this is the case we should probably delete the v2....

Reply:
v2 supports both by way of having both the `/eth/v2/beacon/blocks` and `/eth/v2/beacon/blinded_blocks` endpoints, v3 just unifies the two

Reply:
ummm actually i also cant understand that why have v2 when we can extend v1 (its simply rename v2 to v1 in the codebase, there is no conceptual diff) and its still backward compatible... 

even for publish v2 could have been done away with by just extended v1 ....

Reply:
v2 was only created for deneb... if its causing us to keep a container we don't need, thats sub-optimal...

Reply:
theoretically, the VC doesn't even need access to the blobs. it can just sign the block and let the blob be handled by the BN, as in, creating the `BlobSidecar` from `SignedBeaconBlock` + `KzgProofs / Blobs`

Reply:
Actually, it needs access, because the BN that they are sent to may not have the blobs anymore, if they were originally fetched from a different one.

Reply:
right, thats the idea of statelessness

Reply:
> v2 was only created for deneb... if its causing us to keep a container we don't need, thats sub-optimal...

right I think I missed your point. Agree we should deprecate V2

Reply:
Just wanted to mention that keeping the blobs in the unblinded flow response makes it really easy to update our testing tools for the blob p2p scenarios in testnets.